.github/workflows/docker_build.yml

```diff
@@ -0,0 +1,49 @@
+name: Create and publish a Docker image
+
+on:
+  workflow_call:
+    inputs:
+      NOTEBOOK_TYPE:
+        required: true
+        type: string
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_PATH: ghcr.io/${{ github.repository }}-${{ inputs.NOTEBOOK_TYPE }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2.1.0
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v4.1.1
+        with:
+          images: ${{ env.IMAGE_PATH }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v3.2.0
+        with:
+          context: .
+          build-args: notebook_type=${{ inputs.NOTEBOOK_TYPE }}
+          # Note we only publish containers on pushes to the main branch
+          push: |
+            ${{ github.event_name != 'pull_request' &&
+            (github.ref == 'refs/heads/*' ||
+            github.ref == 'refs/tags/v[0-9]+\\.[0-9]+\\.[0-9]+' )}}
```

Comment:
IMO it would be simpler and easier to understand if you move these conditions to main.yml